uwblueprint/don-efficace
# expose local web server using ngrok

Notion Ticket Link:

https://www.notion.so/uwblueprintexecs/e2d9163a2f8c436bbb36172066753d40?v=9518aa1af04b4478b3d5e840f4ffcc7f&p=850e4246e7d74065826f1de8203d5457&pm=s

Summary:
- Added ngrok implementation so devs have a public URL to hit
- Tested backend endpoints with cURL
- Will install ngrok on docker compose up --build
- Devs must register for their own ngrok account and add their authtoken as an environment variable
- Run ngrok http http://localhost:5001 to open the ngrok interface

Expected result:
1. Rebuilding docker container and running docker compose up --build
2. Run ngrok http http://localhost:5001 to open the ngrok interface (public URL logged in terminal)
3. Public URL should be able to be accessed with cURL GET and POST requests

Ongoing/Incomplete:
* Still need to do a thorough cURL testing of the endpoints (ik how to do them now - will be done by june 5th, 2024)
* Also need to figure out authorization for hitting endpoints

Questions/Notes for Matt:
- In the Dockerfile in the backend / typescript folder, I changed the EXPOSE line to expose 5001 instead of 5000 (line 27)
- It was previously exposing 5000, but that didn't seem to have an effect, so I just changed it to 5001.
- I set most things related to the backend to port 5001 instead of 5000 (see changes)
- I'm not sure how to run a linter (see checklist below)?

## Checklist
- [y] My PR name is descriptive and in imperative tense
- [y] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [?] I have run the appropriate linter(s)
- [y] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
